src/Middleware/OutputCaching/test/MemoryOutputCacheStoreTests.cs

```diff
@@ -140,6 +142,40 @@ public async Task EvictByTag_MultipleTags_MultipleEntries()
         Assert.Null(result2);
     }
 
+    [Fact]
+    public async Task ExpiredEntries_AreRemovedFromTags()
+    {
+        var testClock = new TestMemoryOptionsClock { UtcNow = DateTimeOffset.UtcNow };
+        var store = new MemoryOutputCacheStore(new MemoryCache(new MemoryCacheOptions { SizeLimit = 1000, Clock = testClock, ExpirationScanFrequency = TimeSpan.FromMilliseconds(1) }));
+        var value = "abc"u8.ToArray();
+
+        await store.SetAsync("a", value, new[] { "tag1" }, TimeSpan.FromMilliseconds(5), default);
+        await store.SetAsync("b", value, new[] { "tag1", "tag2" }, TimeSpan.FromMilliseconds(5), default);
+
+        testClock.Advance(TimeSpan.FromMilliseconds(10));
+
+        await store.SetAsync("c", value, new[] { "tag2" }, TimeSpan.FromMilliseconds(10), default);
+        await Task.Delay(10);
```

Comment:
Why do we need this Task.Delay? Because of the threadpool dispatch in InvokeEvictionCallbacks? It feels like this is going to be a source of flakiness.

If there's no good way to synchronize properly in this test, we should probably loop until `resulta` becomes null for the default test timeout.